src/mixins/ExecutorMixin.h

```diff
@@ -19,12 +19,43 @@
 
 namespace reactivesocket {
 
+folly::Executor& defaultExecutor();
+
+class ExecutorBase {
+ public:
+  explicit ExecutorBase(
+      folly::Executor& executor = defaultExecutor(),
+      bool startExecutor = true);
+
+  /// We start in a queueing mode, where it merely queues signal
+  /// deliveries until ::start is invoked.
```

Comment:
I want this base class to be used for the implementation reactive socket automata where we need the functionality. We want to make some calls to the object right after the construction which can eventually call users callback, but we don't want to process user signals until we say start (the instance is fully initialized and ready).
I should write more about this in the documentation.
Good thing is that adds no overhead if not used.